allauth/socialaccount/providers/oauth2/provider.py

```diff
@@ -12,11 +13,12 @@
 from allauth.socialaccount.providers.oauth2.utils import (
     generate_code_challenge,
 )
+from allauth.socialaccount.providers.oauth2.views import OAuth2Adapter
 
 
 class OAuth2Provider(Provider):
     pkce_enabled_default = False
-    oauth2_adapter_class = None
+    oauth2_adapter_class: Optional[Type[OAuth2Adapter]] = None
```

Comment:
This shouldn't be optional.

Reply:
Removed the default and changed (and moved) the instantiation check to raise a real error instead of a warning.